satellite/metainfo/service_test.go

```diff
@@ -55,3 +57,61 @@ func TestIterate(t *testing.T) {
 		require.Equal(t, 1, itemCount)
 	})
 }
+
+func TestUpdatePiecesCheckDuplicates(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 2, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		satellite := planet.Satellites[0]
+		uplink := planet.Uplinks[0]
+		path := "test/path"
+
+		err := uplink.Upload(ctx, satellite, "test1", path, testrand.Bytes(5*memory.KiB))
```

Comment:
We should upload with config to ensure we upload to at least 2 nodes. With a testplanet network of 2 storagenodes, I don't know if a normal `uplink.Upload` will put a piece on both nodes.

Reply:
Updated